builtin/providers/aws/resource_aws_workspace.go

```diff
@@ -0,0 +1,246 @@
+				ForceNew: true,
+				Required: true,
+			},
+
+			"user_volume_encryption": &schema.Schema{
+				Type:     schema.TypeBool,
+				ForceNew: true,
+				Optional: true,
+				Default:  false,
+			},
+
+			"volume_encryption_key": &schema.Schema{
+				Type:     schema.TypeString,
+				ForceNew: true,
+				Optional: true,
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsWorkspaceCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).workspacesconn
+
+	params := workspaces.WorkspaceRequest{
+		DirectoryId: aws.String(d.Get("directory_id").(string)),
+		UserName:    aws.String(d.Get("user_name").(string)),
+	}
+
+	if v, ok := d.GetOk("bundle_id"); ok {
+		params.BundleId = aws.String(v.(string))
+	} else {
+		// Only looking up from the default provided AMAZON bundles at this point
+		foundBundles, err := findWorkspacesBundle("AMAZON", d.Get("bundle_name").(string), meta)
+
+		if err != nil {
+			log.Printf("Error finding bundle: %q", err.Error())
+			return err
+		}
+
+		params.BundleId = foundBundles[0].BundleId
+	}
+
+	if v, ok := d.GetOk("root_volume_encryption"); ok {
+		params.RootVolumeEncryptionEnabled = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("user_volume_encryption"); ok {
+		params.UserVolumeEncryptionEnabled = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("volume_encryption_key"); ok {
+		params.VolumeEncryptionKey = aws.String(v.(string))
+	}
+
+	var ws_request []*workspaces.WorkspaceRequest
+	ws_request[0] = &params
+
+	createOpts := &workspaces.CreateWorkspacesInput{
+		Workspaces: ws_request,
+	}
+
+	// Wait for creation
+	log.Printf("[DEBUG] Waiting for Workspace (%q) to become available", d.Id())
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Pending", "Rebooting", "Rebuilding"},
+		Target:  []string{"Avaliable"},
+		Refresh: func() (interface{}, string, error) {
+			resp, err := conn.CreateWorkspaces(createOpts)
+			if err != nil {
+				log.Printf("Error during creation of Workspace: %q", err.Error())
+				return nil, "", err
+			}
+
+			ws := resp.PendingRequests[0]
+			d.Set("id", *ws.WorkspaceId)
+			log.Printf("[DEBUG] Creation of Workspace %q is in following state: %q.",
+				d.Id(), *ws.State)
+			return ws, *ws.State, nil
+		},
+		Timeout: 30 * time.Minute,
+	}
+
+	if _, err := stateConf.WaitForState(); err != nil {
+		return fmt.Errorf(
+			"Error waiting for Workspace (%s) to become available: %s",
+			d.Id(), err)
+	}
+
+	// Update if we need to
+	return resourceAwsWorkspaceUpdate(d, meta)
+}
+
+func resourceAwsWorkspaceRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).workspacesconn
+
+	resp, err := conn.DescribeWorkspaces(&workspaces.DescribeWorkspacesInput{
+		WorkspaceIds: []*string{aws.String(d.Id())},
+	})
+
+	if err != nil {
+		log.Printf("[ERROR] Error finding Workspace: %s", err)
+		return err
+	}
+
+	// If nothing was found, then return no state
+	if len(resp.Workspaces) == 0 {
+		d.SetId("")
```

Comment:
We should drop in a log.Printf message to say that nothing was found and we are removing from state. This will help people debug why things disappear :)